sending/src/main/java/org/kin/sending/view/ContactDialogFragment.java

```diff
@@ -0,0 +1,220 @@
+            addressInput.clearError();
+            addressNotValidWarning.setVisibility(View.INVISIBLE);
+            presenter.onAddressChanged(editable.toString().trim());
+        }
+    };
+
+    public static ContactDialogFragment getInstance(@NonNull UUID id) {
+        ContactDialogFragment dialog = new ContactDialogFragment();
+        Bundle args = new Bundle();
+        //args.putInt(KEY_STEP, step);
+        args.putSerializable(KEY_ID, id);
+        dialog.setArguments(args);
+        return dialog;
+    }
+
+    public static ContactDialogFragment getInstance() {
+        return new ContactDialogFragment();
+    }
+
+    @Override
+    protected int getLayoutRes() {
+        return R.layout.contact_dialog_fragment;
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        if (!(getActivity() instanceof SendKinView)) {
+            getActivity().finish();
+        }
+        UUID id = null;
+        if (getArguments() != null && getArguments().containsKey(KEY_ID)) {
+            id = (UUID) getArguments().getSerializable(KEY_ID);
+        }
+
+        presenter = new ContactDialogPresenterImpl(((SendKinView) getActivity()).getSendKinPresenter(), id);
+        presenter.onAttach(this);
+        initView(view);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        presenter.onResume();
+    }
+
+    @Override
+    public void setNewLayout() {
+        title.setText(R.string.save_a_new_address);
+        deleteIcon.setVisibility(View.GONE);
+        nameInput.setText("");
+        addressInput.setText("");
+        deleteIcon.setVisibility(View.GONE);
+        addressNotValidWarning.setVisibility(View.INVISIBLE);
+    }
+
+    @Override
+    public void setEditLayout(String name, String address) {
+        title.setText(R.string.edit_address);
+        nameInput.setText(name);
+        addressInput.setText(address);
+        deleteIcon.setVisibility(View.VISIBLE);
+        deleteIcon.setVisibility(View.VISIBLE);
+        addressNotValidWarning.setVisibility(View.INVISIBLE);
+    }
+
+    @Override
+    public void showNameValidity(boolean isValid) {
+        if (isValid) {
+            nameInput.clearError();
+        } else {
+            nameInput.showError();
+        }
+    }
+
+    @Override
+    public void showAddressValidity(boolean isValid, boolean errorDetails) {
+        if (isValid) {
+            addressInput.clearError();
+            addressNotValidWarning.setVisibility(View.INVISIBLE);
+        } else {
+            addressInput.showError();
+            if (errorDetails) {
+                addressNotValidWarning.setVisibility(View.VISIBLE);
+            } else {
+                addressNotValidWarning.setVisibility(View.INVISIBLE);
+            }
+        }
+    }
+
+    @Override
+    public void setConfirmDeleteLayout(@NonNull String nickName) {
```

Comment:
Remove non relevant texts that may appear from the previous dialog(error text for example)

Reply:
good catch